src/main/java/com/edmunds/tools/databricks/maven/BaseDatabricksJobMojo.java

```diff
@@ -71,264 +71,160 @@
+                    getLog().info(String.format("%s|set libraries with %s", jobName, OBJECT_MAPPER.writeValueAsString(defaultSettings.getLibraries())));
+                }
 
-        if (targetDTO.getMaxRetries() == null) {
-            targetDTO.setMaxRetries(defaultDTO.getMaxRetries());
-            getLog().info(String.format("%s|set max_retries with %s", jobName, defaultDTO.getMaxRetries()));
-        }
+                if (settings.getMaxConcurrentRuns() == null) {
+                    settings.setMaxConcurrentRuns(defaultSettings.getMaxConcurrentRuns());
+                    getLog().info(String.format("%s|set max_concurrent_runs with %s", jobName, defaultSettings.getMaxConcurrentRuns()));
+                }
 
-        if (targetDTO.getMaxRetries() != 0 && targetDTO.getMinRetryIntervalMillis() == null) {
-            targetDTO.setMinRetryIntervalMillis(defaultDTO.getMinRetryIntervalMillis());
-            getLog().info(String.format("%s|set min_retry_interval_millis with %s", jobName, defaultDTO.getMinRetryIntervalMillis()));
-        }
+                if (settings.getMaxRetries() == null) {
+                    settings.setMaxRetries(defaultSettings.getMaxRetries());
+                    getLog().info(String.format("%s|set max_retries with %s", jobName, defaultSettings.getMaxRetries()));
+                }
 
+                if (settings.getMaxRetries() != 0 && settings.getMinRetryIntervalMillis() == null) {
+                    settings.setMinRetryIntervalMillis(defaultSettings.getMinRetryIntervalMillis());
+                    getLog().info(String.format("%s|set min_retry_interval_millis with %s", jobName, defaultSettings.getMinRetryIntervalMillis()));
+                }
 
-        //set InstanceTags
-        if (targetDTO.getNewCluster() != null) {
-            String groupId = jobTemplateModel.getGroupWithoutCompany();
-            Map<String, String> tagMap = targetDTO.getNewCluster().getCustomTags();
-            if (tagMap == null) {
-                tagMap = new HashMap();
-                targetDTO.getNewCluster().setCustomTags(tagMap);
+                //set InstanceTags
+                if (settings.getNewCluster() != null) {
+                    String groupId = templateModel.getGroupWithoutCompany();
+                    Map<String, String> tagMap = settings.getNewCluster().getCustomTags();
+                    if (tagMap == null) {
+                        tagMap = new HashMap<>();
+                        settings.getNewCluster().setCustomTags(tagMap);
+                    }
+
+                    if (!tagMap.containsKey(TEAM_TAG) || StringUtils.isEmpty(tagMap.get(TEAM_TAG))) {
+                        tagMap.put(TEAM_TAG, groupId);
+                        getLog().info(String.format("%s|set new_cluster.custom_tags.%s from [%s] to [%s]", jobName, TEAM_TAG, tagMap.get(TEAM_TAG), groupId));
+                    }
+
+
+                    if (ValidationUtil.isDeltaEnabled(settings.getNewCluster())
```

Comment:
to be honest this part should be deleted, but that could be done as part of a separate review.

Reply:
Agree, let's do it within a separate commit